### PR TITLE
Revamp transforms doc

### PR DESCRIPTION
**Pull request by NicolasHug**
- Reorganize API ref in to 2 separate sections: one for V2, one for V1
- Document the v2 functional (docstrings still to be done)
- Lots of re-writes and additional info
- Make v2 the "officially recommended" namespace

Will follow-up with changes to the gallery examples when this is done.

cc @vfdev-5

### CODE REVIEW

File: gallery/v2_transforms/plot_datapoints.py [hunk 1]
```diff
@@ -274,13 +275,13 @@ def get_transform(train):
...
 #
-# 1. Operations like :meth:`~torch.Tensor.clone`, :meth:`~torch.Tensor.to`,
-#    :meth:`torch.Tensor.detach` and :meth:`~torch.Tensor.requires_grad_` retain
-#    the datapoint type.
-# 2. Inplace operations on datapoints like ``.add_()`` preserve they type. However,
-#    the **returned** value of inplace operations will be unwrapped into a pure
-#    tensor:
+# Inplace operations on datapoints like ``obj.add_()`` will preserve the type of
+# ``obj``. However, the **returned** value of inplace operations will be a pure
+# tensor:
```
Comment (NicolasHug) [inline gallery/v2_transforms/plot_datapoints.py:284]: This is more of a drive-by. Turns out those `inplace` operations aren't really exceptions at all.

File: docs/source/transforms.rst [hunk 2]
```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
 
 .. currentmodule:: torchvision.transforms
 
+Torchvision supports common computer vision transformations in the
```
Comment (NicolasHug) [inline docs/source/transforms.rst:8]: All the text below is mostly new. There was very little copy/pasting. Mostly re-writing from scratch.

File: docs/source/transforms.rst [hunk 3]
```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
...
     :template: class.rst
 
-    Resize
     v2.Resize
     v2.ScaleJitter
     v2.RandomShortestSize
     v2.RandomResize
-    RandomCrop
+
+Functionals
```
Comment (NicolasHug) [inline docs/source/transforms.rst:258]: I opted to separate the functionals into [subsubsubsub]sections like that. The alternative is to put e.g. `resize` just below `Resize`, but I found that to hurt readability of the rendered docs because it creates tables that are too long, and it becomes hard to figure out which transforms are actually supported.

(I didn't change the way we document the v1 stuff)

File: docs/source/transforms.rst [hunk 4]
```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
...
+- Resize with bilinear or bicubic mode
+
+This is what a typical transform pipeline could look like:
+
+.. code:: python
+
+    from torchvision.transforms import v2
+    transforms = v2.Compose([
+        v2.ToImage(),  # Convert to tensor, only needed if you had a PIL image
+        v2.ToDtype(torch.uint8, scale=True),  # optional, most input are already uint8 at this point
```
Comment (vfdev-5) [inline docs/source/transforms.rst:143]: I agree that this is optional however I wonder when adding this line makes sense?
If input PIL image has mode "F", than rescaling to uint8 can lead to some data loss ...

Comment (NicolasHug) [reply docs/source/transforms.rst:143]: We generally don't control what user input will be

**Comment by vfdev-5** [comment]
@NicolasHug great work rewriting transforms.rst !

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks a lot !

**Review by pmeier — APPROVED** [review]
Some minor comments inline. Otherwise, I stumbled over this:

https://github.com/pytorch/vision/blob/2c44ebaeece31b0cc9a7385e406312f741333ab5/torchvision/transforms/v2/_type_conversion.py#L46-L47

This does not come from this PR, but might be worth fixing here.

For the most common case, i.e. tensor input and omitting the `mode` parameter, we do in fact scale the values:

https://github.com/pytorch/vision/blob/2c44ebaeece31b0cc9a7385e406312f741333ab5/torchvision/transforms/functional.py#L289-L291

Meaning, this transform will fail for users that do `ToPILImage(my_float_image_tensor * 255)`

Anyway, that is not really important. Thanks Nicolas for the major rewrite!

File: docs/source/transforms.rst [hunk 5]
```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
...
-        import torch
-        torch.manual_seed(17)
+.. note::
+
+    Random transforms like :class:`~torchvision.transforms.v2.RandomCrop` will
+    randomly sample some parameter each time they're called. Their functional
+    counterpart (:func:`~torchvision.transforms.v2.functional.crop`) do not do
+    any kind of random sampling and thus have a slighlty different
+    parametrization. The ``get_params()`` class method of the transforms class
+    can be used to perform parameter sampling when using the functional APIs.
```
[the remark this answers is not in the document]
Comment (NicolasHug) [reply docs/source/transforms.rst:190]: It is still the only way to sample parameters, right? I don't think we just added it for BC, we added it for feature completeness.

File: docs/source/transforms.rst [hunk 6]
```diff
@@ -5,242 +5,449 @@ Transforming and augmenting images
...
+        v2.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    ])
+
+The above should give you the best performance in a typical training environment
+that relies on the :class:`torch.utils.data.DataLoader` with ``num_workers >
+0``.
+
+Transforms tend to be sensitive to the input strides / memory layout. Some
+transforms will be faster with channels-first images while others prefer
+channels-last. You may want to experiment a bit if you're chasing the very
```
[the remark this answers is not in the document]
Comment (NicolasHug) [reply docs/source/transforms.rst:157]: addressed in [7f6a39d](https://github.com/pytorch/vision/pull/7859/commits/7f6a39dc6fc9e0dbacf58f62f8198a00a254591c)
